pkgs/development/libraries/re2/default.nix

```diff
@@ -21,48 +19,51 @@ stdenv.mkDerivation rec {
     owner = "google";
     repo = "re2";
     rev = version;
-    sha256 = "sha256-UontAjOXpnPcOgoFHjf+1WSbCR7h58/U7nn4meT200Y=";
+    hash = "sha256-UontAjOXpnPcOgoFHjf+1WSbCR7h58/U7nn4meT200Y=";
   };
 
-  preConfigure = ''
-    substituteInPlace Makefile --replace "/usr/local" "$out"
-    # we're using gnu sed, even on darwin
-    substituteInPlace Makefile  --replace "SED_INPLACE=sed -i '''" "SED_INPLACE=sed -i"
-  '';
+  outputs = [ "out" "dev" ];
 
-  buildFlags = lib.optionals stdenv.hostPlatform.isStatic [ "static" ];
+  nativeBuildInputs = [ cmake ninja ];
 
-  enableParallelBuilding = true;
+  postPatch = ''
+    substituteInPlace re2Config.cmake.in \
+      --replace "\''${PACKAGE_PREFIX_DIR}/" ""
+  '';
 
-  preCheck = "patchShebangs runtests";
-  doCheck = true;
-  checkTarget = if stdenv.hostPlatform.isStatic then "static-test" else "test";
+  cmakeFlags = [
+    "-DBUILD_SHARED_LIBS:BOOL=${if enableShared then "OFF" else "ON"}"
+  ];
 
-  installTargets = lib.optionals stdenv.hostPlatform.isStatic [ "static-install" ];
+  # This installs a pkg-config definition.
+  postInstall = ''
+    cd "$src"
+    make prefix="$dev" common-install
+  '';
 
-  doInstallCheck = true;
-  installCheckTarget = if stdenv.hostPlatform.isStatic then "static-testinstall" else "testinstall";
+  doCheck = true;
 
-  passthru = {
-    updateScript = nix-update-script {
-      attrPath = pname;
-    };
-    tests = {
-      inherit
-        chromium
-        grpc
-        mercurial;
-      inherit (python3.pkgs)
-        fb-re2
-        google-re2;
-      haskellPackages-re2 = haskellPackages.re2;
-    };
+  passthru.tests = {
+    inherit
+      chromium
+      grpc
+      mercurial;
+    inherit (python3Packages)
+      fb-re2
+      google-re2;
+    haskell-re2 = haskellPackages.re2;
   };
```

Comment:
Do we really need to add these build tests here? Isn't it implied that we are going to rebuild these packages anyway given that the output hash changes?

Reply:
Usually yes, I think the intention was that because of mass-rebuilds it's hard to spot important packages that have r2 as a direct dependency and building all packages with `nixpkgs-review` is not feasible because of the amount of rebuilds.